test-tools/core/src/main/java/org/kie/cloud/tests/core/resources/Project.java

```diff
@@ -0,0 +1,335 @@
+        this.openShift = OpenShifts.master(projectName);
+        this.openShiftAdmin = OpenShifts.admin(projectName);
+    }
+
+    public static final Project getOrCreate(String projectName) {
+        OpenShift openShift = OpenShifts.master();
+        if (openShift.getProject(projectName) == null) {
+            LOGGER.info("Create project {}", projectName);
+            openShift.createProjectRequest(projectName);
+            new SimpleWaiter(() -> openShift.getProject(projectName) != null)
+                    .reason("Waiting for project " + projectName + " to be created.")
+                    .timeout(TimeUnit.MINUTES, 5)
+                    .waitFor();
+        }
+        return new Project(projectName);
+    }
+
+    /**
+     * @return Project name.
+     */
+    public String getName() {
+        return projectName;
+    }
+
+    /**
+     * @return OpenShift client.
+     */
+    public OpenShift getOpenShift() {
+        return openShift;
+    }
+
+    /**
+     * @return OpenShift admin client.
+     */
+    public OpenShift getOpenShiftAdmin() {
+        return openShiftAdmin;
+    }
+
+    /**
+     * Delete OpenShift project.
+     */
+    public void delete() {
+        openShift.deleteProject();
+    }
+
+    /**
+     * Process template and create all resources defined there.
+     *
+     * @param templateUrl URL of template to be processed
+     * @param envVariables Map of environment variables to override default values from the template
+     */
+    public void processTemplateAndCreateResources(URL templateUrl, Map<String, String> envVariables) {
+        boolean templateIsFile = templateUrl.getProtocol().equals("file");
+
+        // Used to log into OpenShift
+        OpenShiftBinary oc = openShiftBinaryClient();
+
+        List<String> commandParameters = new ArrayList<>();
+        commandParameters.add(openShiftBinaryPath());
+        commandParameters.add("process");
+        commandParameters.add("-f");
+        commandParameters.add(templateIsFile ? templateUrl.getPath() : templateUrl.toExternalForm());
+        commandParameters.add("--local");
+        commandParameters.add("--ignore-unknown-parameters=true");
+        commandParameters.add("-o");
+        commandParameters.add("yaml");
+        for (Entry<String, String> envVariable : envVariables.entrySet()) {
+            commandParameters.add("-p");
+            commandParameters.add(envVariable.getKey() + "=" + envVariable.getValue());
+        }
+        String completeProcessingCommand = commandParameters.stream().collect(Collectors.joining(" "));
+
+        try (ProcessExecutor executor = new ProcessExecutor()) {
+            File processedTemplate = executor.executeProcessCommandToTempFile(completeProcessingCommand);
+            oc.execute("create", "-n", getName(), "-f", processedTemplate.getAbsolutePath());
+            //            openShift.load(Files.newInputStream(processedTemplate.toPath()))
+            //                .;
```

Comment:
please, delete these lines

Reply:
will update them. wanted a stable commit but this work is not finished. Need some work based on [previous comment](https://github.com/kiegroup/kie-cloud-tools/pull/8#discussion_r321614664) from @sutaakar